src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -701,21 +726,39 @@ impl ToBatch for LogicalAgg {
 
 impl ToStream for LogicalAgg {
     fn to_stream(&self) -> Result<PlanRef> {
+        let input = self.input();
+        // simple-agg
         if self.group_keys().is_empty() {
-            Ok(StreamSimpleAgg::new(
-                self.clone_with_input(
-                    self.input()
-                        .to_stream_with_dist_required(&RequiredDist::single())?,
-                ),
-            )
-            .into())
+            let has_min_or_max_agg_calls = self
```

Comment:
I'd prefer to explicitly write supported two-phase aggs instead of writing unsupported one, because we will never know whether newly-added aggs can be two-phase.

Maybe:

```
let use_two_phase_agg = xxxx.iter().all(|c| matches!(c.agg_kind, AggKind::Count | AggKind::Sum)));
```